components/infobox/wikis/brawlhalla/infobox_person_player_custom.lua

```diff
@@ -0,0 +1,144 @@
+---
+-- @Liquipedia
+-- wiki=brawlhalla
+-- page=Module:Infobox/Person/Player/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Player = require('Module:Infobox/Person')
+local Class = require('Module:Class')
+local Variables = require('Module:Variables')
+local Math = require('Module:Math')
+local Earnings = require('Module:Earnings')
+local ActiveYears = require('Module:YearsActive')
+local PlayersSignatureLegends = require('Module:PlayersSignatureLegends')
+
+local _PAGENAME = mw.title.getCurrentTitle().prefixedText
+
+--role stuff tables
+local _ROLES = {
+	['admin'] = 'Admin', ['analyst'] = 'Analyst', ['coach'] = 'Coach',
+	['commentator'] = 'Commentator', ['caster'] = 'Commentator',
+	['expert'] = 'Analyst', ['host'] = 'Host', ['streamer'] = 'Streamer',
+	['interviewer'] = 'Interviewer', ['journalist'] = 'Journalist',
+	['manager'] = 'Manager', ['map maker'] = 'Map maker',
+	['observer'] = 'Observer', ['photographer'] = 'Photographer',
+	['tournament organizer'] = 'Organizer', ['organizer'] = 'Organizer',
+}
+local _CLEAN_OTHER_ROLES = {
+	['coach'] = 'Coach', ['staff'] = 'false',
+	['content producer'] = 'Content producer', ['streamer'] = 'false',
+}
+
+local _CURRENT_YEAR = tonumber(os.date('%Y'))
+local _statusStore
+
+local Injector = require('Module:Infobox/Widget/Injector')
+local Cell = require('Module:Infobox/Widget/Cell')
+
+local CustomPlayer = Class.new()
+
+local CustomInjector = Class.new(Injector)
+
+local _args
+local _earningsPerYear = {}
+
+function CustomPlayer.run(frame)
+	local player = Player(frame)
+	_args = player.args
+
+	player.getStatusToStore = CustomPlayer.getStatusToStore
+	player.adjustLPDB = CustomPlayer.adjustLPDB
+	player.getPersonType = CustomPlayer.getPersonType
+
+	player.calculateEarnings = CustomPlayer.calculateEarnings
+	player.createWidgetInjector = CustomPlayer.createWidgetInjector
+
+	return player:createInfobox(frame)
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'role' then return {}
+	elseif
+		id == 'history' and
+		string.match(_args.retired or '', '%d%d%d%d')
+	then
+		table.insert(widgets, Cell{
+				name = 'Retired',
+				content = {_args.retired}
+			})
+	end
+	return widgets
+end
+
+function CustomInjector:addCustomCells(widgets)
+	local yearsActive = ActiveYears.display{
+		player = _PAGENAME,
+	}
+
+	local currentYearEarnings = _earningsPerYear[tostring(_CURRENT_YEAR)]
+	if currentYearEarnings then
+		currentYearEarnings = Math.round{currentYearEarnings}
+		currentYearEarnings = '$' .. mw.language.new('en'):formatNum(currentYearEarnings)
+	end
```

Comment:
Any perticual reason why the commons earnings can't be used?
(`_earningsPerYear`)

Reply:
how would i access `_earningsPerYear` (which is set in the commons part) in the /Custom part?

Reply:
Ah yeah, they're set as local variables instead of class variables.

Reply:
I'm fine with making it a class/instance variable instead of a local variable.

Reply:
#1200